t/chaos/kill-etcd_test.go

```diff
@@ -0,0 +1,257 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package chaos
+
+import (
+	"fmt"
+	"net/http"
+	"os/exec"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/gavv/httpexpect/v2"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	token = "edd1c9f034335f136f87ad84b625c8f1"
+	host  = "http://127.0.0.1:9080"
+)
+
+type httpTestCase struct {
+	E            *httpexpect.Expect
+	Method       string
+	Path         string
+	Body         string
+	Headers      map[string]string
+	ExpectStatus int
+	ExpectBody   string
+}
+
+func caseCheck(tc httpTestCase) *httpexpect.Response {
+	e := tc.E
+	var req *httpexpect.Request
+	switch tc.Method {
+	case http.MethodGet:
+		req = e.GET(tc.Path)
+	case http.MethodPut:
+		req = e.PUT(tc.Path)
+	default:
+	}
+
+	if req == nil {
+		fmt.Printf("%#v", tc)
+		panic("fail to init request")
+	}
+	for key, val := range tc.Headers {
+		req.WithHeader(key, val)
+	}
+	if tc.Body != "" {
+		req.WithText(tc.Body)
+	}
+
+	resp := req.Expect()
+	if tc.ExpectStatus != 0 {
+		resp.Status(tc.ExpectStatus)
+	}
+
+	if tc.ExpectBody != "" {
+		resp.Body().Contains(tc.ExpectBody)
+	}
+
+	return resp
+}
+
+func setRoute(e *httpexpect.Expect, expectStatus int) {
+	caseCheck(httpTestCase{
+		E:       e,
+		Method:  http.MethodPut,
+		Path:    "/apisix/admin/routes/1",
+		Headers: map[string]string{"X-API-KEY": token},
+		Body: `{
+			"uri": "/hello",
+			"host": "foo.com",
+			"plugins": {
+				"prometheus": {}
+			},
+			"upstream": {
+				"nodes": {
+					"bar.org": 1
+				},
+				"type": "roundrobin"
+			}
+		}`,
+		ExpectStatus: expectStatus,
+	})
+}
+
+func getRoute(e *httpexpect.Expect, expectStatus int) {
+	caseCheck(httpTestCase{
+		E:            e,
+		Method:       http.MethodGet,
+		Path:         "/hello",
+		Headers:      map[string]string{"Host": "foo.com"},
+		ExpectStatus: expectStatus,
+	})
+}
+
+func deleteRoute(e *httpexpect.Expect, expectStatus int) {
+	caseCheck(httpTestCase{
+		E:            e,
+		Method:       http.MethodDelete,
+		Path:         "/apisix/admin/routes/1",
+		Headers:      map[string]string{"X-API-KEY": token},
+		ExpectStatus: expectStatus,
+	})
+}
+
+/*
+// not working, don't know why
```

Comment:
Can we remove the commented code?

Reply:
Sure I'll leave that to a new PR

Reply:
Commented code removed, and the question is raised in https://github.com/apache/apisix-dashboard/issues/1388